src/modules/Common/private/Get-SdnRole.ps1

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+function Get-SdnRole {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true)]
+        [System.String]$ComputerName
+    )
+
+    if ($null -eq $Global:SdnDiagnostics.EnvironmentInfo) {
+        throw New-Object System.NullReferenceException("Unable to enumerate data from EnvironmentInfo")
+    }
+
+    foreach ($role in ($Global:SdnDiagnostics.EnvironmentInfo.Keys | Where-Object {$_ -iin $Global:SdnDiagnostics.Config.Keys})) {
+        if ($ComputerName -iin $Global:SdnDiagnostics.EnvironmentInfo[$role]) {
```

Comment:
Will this handle $ComputerName in both ShortName and FQDN format?

Reply:
Good catch. I need to add this to the logic detection to handle FQDN vs just NetBIOS.

Reply:
Pushed a new code change to handle this.